multiapps-controller-web/src/main/java/org/cloudfoundry/multiapps/controller/web/configuration/CsrfAccessDeniedHandler.java

```diff
@@ -1,30 +1,24 @@
-package org.cloudfoundry.multiapps.controller.web.security;
+package org.cloudfoundry.multiapps.controller.web.configuration;
 
-import java.io.IOException;
-
-import javax.inject.Named;
-import javax.servlet.ServletException;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import org.apache.http.HttpStatus;
 import org.cloudfoundry.multiapps.controller.web.Constants;
 import org.springframework.security.access.AccessDeniedException;
-import org.springframework.security.oauth2.provider.error.OAuth2AccessDeniedHandler;
+import org.springframework.security.web.access.AccessDeniedHandler;
 import org.springframework.security.web.csrf.InvalidCsrfTokenException;
 import org.springframework.security.web.csrf.MissingCsrfTokenException;
 
-@Named("accessDeniedHandler")
-public class CsrfAccessDeniedHandler extends OAuth2AccessDeniedHandler {
+public class CsrfAccessDeniedHandler implements AccessDeniedHandler {
 
     private static final String CSRF_TOKEN_REQUIRED_HEADER_VALUE = "Required";
 
     @Override
-    public void handle(HttpServletRequest request, HttpServletResponse response, AccessDeniedException authException)
-        throws IOException, ServletException {
-        if (authException instanceof InvalidCsrfTokenException || authException instanceof MissingCsrfTokenException) {
+    public void handle(HttpServletRequest request, HttpServletResponse response, AccessDeniedException accessDeniedException) {
+        if (accessDeniedException instanceof InvalidCsrfTokenException || accessDeniedException instanceof MissingCsrfTokenException) {
             response.setHeader(Constants.CSRF_TOKEN, CSRF_TOKEN_REQUIRED_HEADER_VALUE);
+            response.setStatus(HttpStatus.SC_FORBIDDEN);
         }
-        doHandle(request, response, authException);
```

Comment:
don`t we need this?

Reply:
Not anymore, now AccessDeniedHandler is implemented